### PR TITLE
Change `Uuid` column name to `Id`

### PR DESCRIPTION
**Pull request by GiaJordan**
## What Changed
This PR makes the switch from naming the (former) `Uuid` column containing UUID values to `Id` instead.

### Manifests
Manifests that contain columns named `Uuid` will have the column name updated to `Id` at upload, and the `.csv` file will be saved with the new convention, **overwriting the version with the old convention**.

### Tables
Tables that have `Uuid` columns will have the column name updated at the next manifest upload whether using the `replace` or `upsert` table operations. All new tables created going forward will be using the `Id` naming convention. Functionality is included to handle cases where manifests and tables have mismatched naming conventions; any adhering to the old convention will be updated to the new `Id` convention.

Tested with example data model manifests and HTAN model manifests

## To Test

1. `git checkout develop`
2. Generate a **new** manifest. If a manifest already exists in the dataset folder then delete and regenerate
    * `Biospecimen` for HTAN entity based, table creation or replacement
        * Can use dataset `syn51398208`
    *  `ScRNASeqLevel1` for HTAN file based, table creation or replacement
        * Can use dataset `syn51398808`
    *  `MockRDB` for example data model, table upserts
        * Can use dataset `syn51470395`
4. If using provided datasets in project `syn27042964`, then delete `Biospecimen`, `ScRNA-SeqLevel1`, and `MockRDB` tables
5. Submit as table with `--table_manipulation` set appropriately
6. `git checkout develop-ID-column-FDS-70` 
7. Add row to manifest; do not change column names
8. Submit as table with `--table_manipulation` set appropriately
9. Verify table reflects all rows in the manifest, with the `Uuid` column now named `Id`
10. Verify local copy of the manifest has had the `Uuid` column changed to be named `Id`

### CODE REVIEW

**Comment by mialy-defelice** [comment]
@GiaJordan Ill start looking through this PR today. Sorry for the delay!

File: schematic/store/synapse.py
Comment (mialy-defelice) [inline schematic/store/synapse.py:2045]: @GiaJordan I know this would be annoying but I think we should check the model to see if the model itself is generating a UUID column before deleting, just as a precaution.

**Comment by linglp** [comment]
@GiaJordan I also tested this PR out of curiosity. I am wondering if it will be worth to modify`test_upsertTable` function so that it checks if `Id` column exists?

File: schematic/store/synapse.py
Comment (mialy-defelice) [reply schematic/store/synapse.py:2045]: Yes I think so. I don't think we want to ban people from having a Uuid column in case they feel like they need one for whatever reason.

**Comment by mialy-defelice** [comment]
Btw @GiaJordan I really love the completeness of the description you provided for this PR, its super helpful for the testing :)

**Comment by mialy-defelice** [comment]
@GiaJordan can you also make sure SchemaHub is also updated to reflect these changes? We can create a new issue to track that work too.

@AmyHeiser for visibility per our earlier meeting on Github PR discussions.

**Comment by milen-sage** [comment]
@AmyHeiser here's an example of the conversations and PR descriptions which are on github but would be very useful for context. We wouldn't want to move them away from github since the code is there and references in the conversations and PR descriptions.

File: schematic/store/synapse.py
Comment (GiaJordan) [reply schematic/store/synapse.py:2045]: @mialy-defelice do you know what happens now if the schema is generating a `Uuid` column? From what I see it only creates a new `Uuid` column if one isn't present, and only fills in uuid values if there isnt something already there. Is that correct?

Comment (GiaJordan) [reply schematic/store/synapse.py]: @mialy-defelice is there a way to see if a given column is in the schema besides the `is_class_in_schema` method from the schema explorer?

File: schematic/store/synapse.py
Comment (GiaJordan) [inline schematic/store/synapse.py:2192]: switch to use `col.name` to handle properties cases
per @mialy-defelice

**Comment by mialy-defelice** [comment]
Hey @GiaJordan,

I have been testing and am getting an odd error. Was I supposed to also install `schematic_db`? I did `poetry install` but am not seeing `schematic_db` in it.

**Here is what I did:**

- I deleted the `Biospecimen` manifest table.
- Added it back with the following: 

```
schematic model --config config.yml submit -mp tests/data/1211/biospecimen_manifest.csv -d syn27042964 -mrt table_and_file -tm upsert
```

- I added an extra row then tried to submit it again.
- I then get the following error:

```
(schematicpy-py3.9) sh-3.2# schematic model --config config.yml submit -mp tests/data/1211/biospecimen_manifest.csv -d syn27042964 -mrt table_and_file -tm upsert
Starting schematic...
The (model > input > location) argument with value 'tests/data/example.model.jsonld' is being read from the config file.
The (model > input > file_type) argument with value 'local' is being read from the config file.
 [####################]100.00%   1/1   Done...
Downloading  [####################]100.00%   17.0kB/17.0kB (11.8MB/s) SYNAPSE_TABLE_QUERY_124884133.csv Done...
 [####################]100.00%   1/1   Done...
Downloading  [####################]100.00%   17.0kB/17.0kB (250.4kB/s) SYNAPSE_TABLE_QUERY_124884136.csv Done...
 [####################]100.00%   1/1   Done...
Downloading  [####################]100.00%   17.0kB/17.0kB (37.1MB/s) SYNAPSE_TABLE_QUERY_124884142.csv Done...
 [####################]100.00%   1/1   Done...
Downloading  [####################]100.00%   17.0kB/17.0kB (253.6kB/s) SYNAPSE_TABLE_QUERY_124884144.csv Done...
    Welcome, Mialy DeFelice!

Welcome, Mialy DeFelice!

Traceback (most recent call last):
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/bin/schematic", line 6, in <module>
    sys.exit(main())
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/click/decorators.py", line 38, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/models/commands.py", line 122, in submit_manifest
    manifest_id = metadata_model.submit_metadata_manifest(
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/models/metadata.py", line 382, in submit_metadata_manifest
    manifest_id = syn_store.associateMetadataWithFiles(
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 1610, in associateMetadataWithFiles
    manifest_synapse_file_id = self.upload_manifest_as_table(
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 1411, in upload_manifest_as_table
    manifest_synapse_table_id, manifest, table_manifest = self.uploadDB(
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 290, in wrapper
    return method(*args, **kwargs)
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 916, in uploadDB
    manifest_table_id = self.buildDB(datasetId, table_name, col_schema, table_manifest, table_manipulation, sg, restrict,)
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 1007, in buildDB
    manifest_table_id = TableOperations.upsertTable(self, sg=sg, tableToLoad = table_manifest, tableName=table_name, existingTableId=table_info[table_name], datasetId=datasetId)
  File "/Users/mialydefelice/Documents/schematic_e/schematic/schematic/store/synapse.py", line 2152, in upsertTable
    synapseDB.upsert_table_rows(table_name=tableName, data=tableToLoad)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/schematic_db/rdb/synapse_database.py", line 426, in upsert_table_rows
    self._upsert_table_rows(table_id, data, primary_key)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/schematic_db/rdb/synapse_database.py", line 438, in _upsert_table_rows
    table = self._create_primary_key_table(table_id, primary_key)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/schematic_db/rdb/synapse_database.py", line 478, in _create_primary_key_table
    table = table[["ROW_ID", "ROW_VERSION", primary_key]]
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/pandas/core/frame.py", line 3813, in __getitem__
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 6070, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/Users/mialydefelice/Library/Caches/pypoetry/virtualenvs/schematicpy-rs9bwhSl-py3.9/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 6133, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['Biospecimen_id'] not in index"
```

**Comment by GiaJordan** [comment]
Hey @mialy-defelice, I think what's happening is that the code is looking for the `Biospecimen_id` attribute/column in the manifest and it isn't finding it. 
The `Biospecimen` component isn't set up for upserts right now, so you should only try to use `--table_manipulation replace` or creating a new table with it. The `MockRDB` component/manifest will work for testing upserts however.

**Review by mialy-defelice — APPROVED** [review]

**Comment by GiaJordan** [comment]
@mialy-defelice thanks for testing!

**Comment by GiaJordan** [comment]
Adding performance metrics to compare current column solution (copy create new, copy over values, delete old) to suggestion to use synapse API requests to rename column
<table>
<thead>
<tr>
<th>Performance Time (seconds)</th>
<th>CLI</th>
<th>API</th>
</thead>
<tbody>
<tr>
<td>Adding 200 rows to a table with 1 existing row</td>
<td>47</td>
<td>116</td>
</tr>
<tr>
<td>Adding 200 rows to a table with ~200 existing rows</td>
<td>41</td>
<td>148</td>
</tr>
</tbody>

**Comment by GiaJordan** [comment]
Based on conversations with @AmyHeiser and others, we'll be waiting until #1229 is merged into this branch before merging this into `develop`

**Comment by GiaJordan** [comment]
> Based on conversations with @AmyHeiser and others, we'll be waiting until #1229 is merged into this branch before merging this into `develop`

Updating after Code review, we'll be merging to better manage other existing and in progress PRs

**Pull request merged**
